src/IGCompiler.ts

```
@@ -0,0 +1,168 @@
+const readDirPmd = util.promisify(readdir);
+const writeFilePmd = util.promisify(writeFile);
+
+const BASE_FHIR_NAME = 'hl7.fhir.r4.core';
+
+export async function loadJson(fileName: PathLike): Promise<any> {
+    return JSON.parse(await readFilePmd(fileName, 'utf8'));
+}
+
+export async function storeJson(fileName: PathLike, data: any) {
+    await writeFilePmd(fileName, JSON.stringify(data));
+}
+
+async function listIgDirs(parentDir: PathLike): Promise<string[]> {
+    return (await readDirPmd(parentDir, { withFileTypes: true }))
+        .filter(dirent => {
+            return (
+                dirent.isDirectory() ||
+                (dirent.isSymbolicLink() &&
+                    statSync(realpathSync(path.join(parentDir.toString(), dirent.name))).isDirectory())
+            );
+        })
+        .map(dirent => {
+            return path.join(parentDir.toString(), dirent.name);
+        });
+}
+
+export interface IGCompilerOptions {
+    ignoreVersion: boolean;
+}
+
+export interface IGInfo {
+    id: string;
+    name: string;
+    version: string;
+    url: string;
+    path: string;
+    dependencies: string[];
+}
+
+/**
+ *  Helper class used for compiling IGs. Its main functions are:
+ *  - Looks through a folder of IG packs
+ *  - Make sure no dependencies are missing
+ *  - and there are no circular dependencies
+ *  - calls compile function for each SearchParameters
+ *
+ */
+export class IGCompiler {
+    private options: IGCompilerOptions;
+
+    private readonly implementationGuides: ImplementationGuides;
+
+    constructor(implementationGuides: ImplementationGuides, options: IGCompilerOptions) {
+        this.options = options;
+        this.implementationGuides = implementationGuides;
+    }
+
+    async collectResources(igDir: PathLike, resources: any[]): Promise<void> {
+        const indexJson = path.join(igDir.toString(), '.index.json');
+        if (!existsSync(indexJson)) {
+            return;
```

I think we should throw an error here. 

This condition will rarely happen, but it would mean that the IG cannot be processed. It'd be better to fail the compilation instead of skipping an entire IG silently.

---

For testing it was convenient to point the compiler to a node_modules folder where some IGs were installed next to some other dependencies.